### ISSUE TITLE
[MU4 Task] Implement the feature to fix corrupted scores

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#16280]
**Task description**
Add ability to fix broken measures/elements by one click and to get rid of corrupted score dialog

**Comment by DmitryArefiev** [comment]
@Tantacrul It would be a nice feature to get rid of that annoying message :) 
(of course, specifications and design is needed here)

**Comment by DmitryArefiev** [comment]
Oh thanks! Will check this out

**Comment by Jojo-Schmitz** [comment]
I've just ported it to 4.x, untested though

**Comment by DmitryArefiev** [comment]
Yeah, it works in **4.0.2** and it fixed 80% of broken measure in one score (at least Corrupted score dialog>Show details list showed me)
That's cool!

@Tantacrul As a temporary solution, we can include it to shipping plugins with 4.0.2

**Comment by Jojo-Schmitz** [comment]
Just needs a fancy icon ;-)

**Comment by Tantacrul** [comment]
Hmmm. In this case, I'd prefer to just fix the issue. It seems weird to have plugins that do what MuseScore really **should** do itself.

I'm grateful to @Jojo-Schmitz for pointing it out because we can simply lift the solution. I'm going to add this to 4.1

**Comment by cbjeukendrup** [comment]
Just a thought: even if MuseScore could somehow fix corruptions automatically, I think it needs to tell the user that this has happened (and in which measures etc), because still data might be lost, or the fix might not be what the user had in mind. 

(Unless we're 100% certain that MuseScore's fix is correct, of course; but I'm afraid we can't be that certain.)

**Comment by Jojo-Schmitz** [comment]
Of course, the user must be a) told about the corruptions and b) asked whether MuseScore should attempt to fix them

**Comment by Tantacrul** [comment]
> 

Yeah, we should discuss this when it comes time to design it. Perhaps automatically trying to fix would be sensible and (as you've said) we just message it properly.

**Comment by Jojo-Schmitz** [comment]
Probably after fixing it, the check should be run again
